src/components/Map/Map.js

```diff
@@ -541,23 +542,36 @@ export default class Map extends React.Component<Props, State> {
 
     const zoom = overrideZoom || fallbackZoom;
 
-    // use the feature coordinates
     const feature = props.feature;
+
+    let center;
+
+    // Use the feature coordinates
     if (
       feature &&
+      getFeatureId(feature) !== this.lastFeatureId &&
```

Comment:
Could this not lead to weird behavior if the feature id did not change? Or are the else ifs not causing any side effects when this is called without a feature change?

Reply:
Not sure if there even is a way to trigger this.

Reply:
I'm not 100% sure what you mean. 😕 

This line guarantees the center is only set to the features coordinates when the marker/feature was first selected. Any future changes in the tree will use the app state center position until a new marker was selected (or the old one simply deselected).

Reply:
The app state center position is updated everytime the map was updated (moveend, zoomend). Also when the marker was clicked and moved into the map center.

Reply:
The behavior in the `else` branches changed. I'll better just come over :D

Reply:
Makes sense!